code/modules/integrated_electronics/core/pins.dm

```diff
@@ -115,7 +115,7 @@ list[](
 		holder.on_data_written()
 	else if(islist(new_data))
 		var/list/new_list = new_data
-		data = new_list.Copy(1,min( IC_MAX_LIST_LENGTH+1, new_list.len ))
+		data = new_list.Copy(max(1,new_list.len - IC_MAX_LIST_LENGTH+1),new_list.len +1)
```

Comment:
the second copy arg can just be 0 if you want the end of the list.

Is it intentional that you're cutting off the front of the list to conform with max list length?

Reply:
yes,it's intentional for some reason.First option don't work.